server/src/controllers/admin.controller.ts

```diff
@@ -0,0 +1,85 @@
+import User from "../models/User/User";
+import Product from "../models/Product/Product";
+
+import express from "express";
+
+const AdminController = {
+  async newProduct(
+    req: express.Request,
+    res: express.Response,
+    next
+  ): Promise<express.Response> {
+    try {
+      const product = await Product.create({ ...req.body });
+      product.save().catch((err) => console.log("Error saving product ", err));
+      return res.status(200).json({
+        product
+      });
+    } catch (err) {
+      return next({
+        status: 400,
+        message: err.message
+      });
+    }
+  },
```

Comment:
Where is this next function passing parameters to? If not required, I think we can remove them from all the controller functions